internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -73,14 +82,36 @@ func (hr *HypershiftAgentServiceConfigReconciler) Reconcile(origCtx context.Cont
 	}
 
 	// Creating spoke client using specified kubeconfig secret reference
-	// TODO: use spoke client to create CRDs/CRs
-	_, err := hr.createSpokeClient(ctx, instance.Spec.KubeconfigSecretRef.Name, instance.Namespace)
+	spokeClient, err := hr.createSpokeClient(ctx, instance.Spec.KubeconfigSecretRef.Name, instance.Namespace)
 	if err != nil {
-		log.WithError(err).Error("Failed to create client using remote kubeconfig", req.NamespacedName)
+		log.WithError(err).Error("Failed to create client using specified kubeconfig", req.NamespacedName)
+		return ctrl.Result{}, err
+	}
+
+	// Ensure agent-install CRDs exist on spoke cluster and clean stale ones
+	if err := hr.syncSpokeAgentInstallCRDs(ctx, spokeClient); err != nil {
+		log.WithError(err).Error(fmt.Sprintf("Failed to sync agent-install CRDs on spoke cluster in namespace '%s'", instance.Namespace))
+		conditionsv1.SetStatusConditionNoHeartbeat(&instance.Status.Conditions, conditionsv1.Condition{
```

Comment:
it's problematic to set single condition, usually when reporting a status you will set all the condition in each reconcile, maybe a different pattern is required here, something similar to what we have in the other controllers that you have a function that update the status with all the conditions.
I'm not sure how it will feet, we can split it into another PR if needed.

Reply:
ok then I'll remove the condition for this PR for now, and add it separately. It would require some work to do it as in cd_ctrl: https://github.com/openshift/assisted-service/blob/ef1a92878d313976b142abf6dbd0909639cfe7f5/internal/controller/controllers/clusterdeployments_controller.go#L1943